mb-nft-v2/src/lib.rs

```diff
@@ -71,17 +70,7 @@ pub struct MintbaseStore {
     /// Token. The key is generated from `tokens_minted`. The map keeps count
     /// of how many copies of this token remain, so that the element may be
     /// dropped when the number reaches zero (ie, when tokens are burnt).
-    #[allow(clippy::type_complexity)] // sorry
-    pub token_metadata: LookupMap<
-        u64,
-        (
-            u16,                    // number of minted tokens
```

Comment:
You do this u32 -> u16 cast later with a comment about "compatibility". Is this backwards compatibility? Or why not use u32 here?

Reply:
This place is no problem. But the `TokenMetadata` struct contains a field `copies: Option<u16>`. If we changed that, all the existing NFT contracts would break. Which just brings me around to that the cast might panic, will have to try this. Might be worth just returning `None` in case of overflow.

Reply:
I think there is a checked_sub function for these kinds of things.

Reply:
Does not panic, typecast simply seems to be a slicing the last significant bits, i.e. `(u16::MAX as u32 + 1) as u16 == 0`

Reply:
Fixed this.